src/vmod_soap_xml.c

```diff
@@ -108,10 +108,20 @@ const char* evaluate_xpath(struct priv_soap_vcl *soap_vcl, struct priv_soap_task
 	for (i = 0; i < (xpathObj->nodesetval ? xpathObj->nodesetval->nodeNr : 0); i++) {
 		if( xpathObj->nodesetval->nodeTab[i]->children &&
 		    xpathObj->nodesetval->nodeTab[i]->children->content ) {
-			return ((const char*)(xpathObj->nodesetval->nodeTab[i]->children->content));
+			// Save free pointer to reset workspace in case of overflow
+			void *f = soap_task->ctx->ws->f;
+			char *res = WS_Copy(soap_task->ctx->ws, xpathObj->nodesetval->nodeTab[i]->children->content, xmlStrlen(xpathObj->nodesetval->nodeTab[i]->children->content) + 1);
+			xmlXPathFreeContext(xpathCtx);
+			xmlXPathFreeObject(xpathObj);
+			if (!res) {
+				WS_Reset(soap_task->ctx->ws, f);
```

Comment:
we don't need to store f or reset it, because WS_Copy doiesn't change it. However we have to call ws_ClearOverflow to not break the WS fort further usage.

Reply:
ws_ClearOverflow declared as 
```
static void
ws_ClearOverflow(struct ws *ws)
```
so it can't be used outside of cache_ws.c. The only way to clear overflow is to call WS_Reset